lib/Target/LLVMIR/LLVMIRTranslation.cpp

```diff
@@ -145,10 +148,47 @@ translateTritonGPUToLLVMIR(llvm::LLVMContext *llvmContext,
     llvm::errs() << "Pass execution failed";
     return nullptr;
   }
+  SmallVector<LLVM::LLVMFuncOp> funcs;
+  module.walk([&](LLVM::LLVMFuncOp func){
+    funcs.push_back(func);
+  });
+
+  static const std::string key("libpath:");
+  std::set<std::string> extern_libs;
+  for (auto &func : funcs) {
+    if (!func.isExternal())
+      continue;
+    auto attrs = func.getPassthroughAttr();
+    for (auto& attr : attrs) {
+      if (StringAttr sAttr = attr.dyn_cast<StringAttr>()) {
+        auto value = sAttr.str();
+        if (0 == value.find(key)) {
+          extern_libs.insert(value.substr(key.size()));
```

Comment:
Instead of parsing the string. Is it possible to instantiate a key->value mapping just as a normal attribute?

For example, can you add the "libpath" attribute to the LLVMFuncOp and get it later?

Reply:
LLVMFuncOp don't have libpath property, and it comes from mlir::llvm::dialect.

Should we modify upstream mlir code?

Reply:
> Should we modify upstream mlir code?

Nope. I meant using the setAttr and getAttr APIs https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a47bdcbd4defd6dd4c2f63fc702d4d84f

Reply:
Thanks, I get your point.